MobileCenter/MobileCenter/Internals/Util/MSUtility+Application.m

```diff
@@ -17,40 +18,46 @@ + (MSApplicationState)applicationState {
 }
 
 #if TARGET_OS_OSX
-+ (NSApplication *)sharedApp {
++ (NSApplication *)sharedApplication {
 
   // Compute selector at runtime for more discretion.
   SEL sharedAppSel = NSSelectorFromString(@"sharedApplication");
   return ((NSApplication * (*)(id, SEL))[[NSApplication class] methodForSelector:sharedAppSel])([NSApplication class],
                                                                                                 sharedAppSel);
 }
 #else
-+ (UIApplication *)sharedApp {
++ (UIApplication *)sharedApplication {
 
   // Compute selector at runtime for more discretion.
   SEL sharedAppSel = NSSelectorFromString(@"sharedApplication");
-  return ((UIApplication * (*)(id, SEL))[[UIApplication class] methodForSelector:sharedAppSel])([UIApplication class],
-                                                                                                sharedAppSel);
+  if ([[UIApplication class] instancesRespondToSelector:sharedAppSel]) {
```

Comment:
Yep a good guard to have, though `UIApplication` always responds to `sharedApplication` even in app extensions. Agree to keep the test in place since `methodForSelector` should always be backed up with `instancesRespondToSelector`. 

There is one thing to fix though, `sharedApplication` is a class method not an instance method so this test will always return `nil`. Should replace by `respondsToSelector:`.